### PR TITLE
Add extension to automatically remove SRI on build

### LINKED ISSUE #19: Regenerate SRI on build
We have JS assets which change during development, but it seems the only way to get the SRI to update is to hit `?updatesri=true` after deployment (e.g. to production). Could there be an easy way of configuring this to regenerate these when a dev/build is called?

Also, probably a separate issue, the `?updatesri=true` only works if you've got a security group with the code "administrators", so it was silently not doing anything for me because we'd changed the groups. There's probably a more reliable way of checking for an appropriate permission, and possibly could be worth throwing an error if you pass that param but don't have permission?

### PR DESCRIPTION
**Pull request by matt-in-a-hat**
If this extension is applied (by adding the example yaml) then running dev/build clears out all SRI and allows them to be recalculated.
Fixes #19

### CODE REVIEW

File: src/Extensions/SRIRefreshExtension.php [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+class SRIRefreshExtension extends DataExtension
+{
+    /**
+     * Deletes the Sub-resource integrity values on build of the database
+     * so they're regenerated next time that file is used.
+     */
+    public function onAfterBuild()
+    {
+        foreach (SRI::get() as $item) {
```
Comment (Firesphere) [inline src/Extensions/SRIRefreshExtension.php:16]: `SRI::get()->deleteAll()` would do the trick.

File: readme.md [hunk 2]
```diff
@@ -127,6 +127,16 @@ To force-refresh the SRI calculations, add the URL Parameter `?updatesri=true`.
 To force the headers to be set in dev-mode, for testing purposes, add the URL parameter `?build-headers=true`.
 To disable this again, change the `true` to `false`
 
+To have these automatically clear out on a dev/build (useful for updating integrity hashes on production when new assets are deployed) you can apply the SRIRefreshExtension by adding the following to your yaml config
```
Comment (Firesphere) [inline readme.md:130]: I feel this would more be a config than requiring an entire extension.

**Review by Firesphere — CHANGES_REQUESTED** [review]

File: src/Models/SRI.php [hunk 3]
```diff
@@ -149,4 +157,18 @@ public function providePermissions()
...
     }
+
+    /**
+     * If configured, this deletes the Sub-resource integrity values on build of the database
+     * so they're regenerated next time that file is used.
+     * Note that this hook only exists in silverstripe-framework 4.7+
+     */
+    public function onAfterBuild()
+    {
+        if ($this->config()->get('clear_sri_on_build')) {
```
Comment (Firesphere) [inline src/Models/SRI.php:168]: Wouldn't `SRI::get()->deleteAll()` do the trick here?

File: src/Tasks/SRIRefreshTask.php [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        these will regenerate on the next request.';
+
+    /**
+     * Deletes the Sub-resource integrity values on build of the database
+     * so they're regenerated next time that file is required.
+     */
+    public function run($request)
+    {
+        echo "Removing SRI values...\n";
+        foreach (SRI::get() as $item) {
```
Comment (Firesphere) [inline src/Tasks/SRIRefreshTask.php:18]: As above, `deleteAll()` should do it

File: src/Tasks/SRIRefreshTask.php [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    protected $description = 'Deletes cache of Sub-resource integrities for js/css resources,
+        these will regenerate on the next request.';
+
+    /**
+     * Deletes the Sub-resource integrity values on build of the database
+     * so they're regenerated next time that file is required.
+     */
+    public function run($request)
+    {
+        echo "Removing SRI values...\n";
```
Comment (Firesphere) [inline src/Tasks/SRIRefreshTask.php:17]: I think a BuildTask has a formatted way of outputting this?

**Review by Firesphere — CHANGES_REQUESTED** [review]

File: src/Models/SRI.php (hunk 3 above)
Comment (matt-in-a-hat) [reply src/Models/SRI.php:168]: DataList only has `removeAll`, and as far as I can see it then loads each from the database individually to delete them. This seems more efficient.

Comment (Firesphere) [reply src/Models/SRI.php:168]: Although in theory, that makes sense, semantically, it makes more sense to me to use deleteAll(). Given this is only done during build, I don't see the need for super performance versus shorter logic

**Comment by Firesphere** [comment]
Could you also please rebase?

**Comment by matt-in-a-hat** [comment]
Rebased and updated to use DataList::removeAll

**Comment by Firesphere** [comment]
Looks good to me. I think it only needs a rebase, so the automated tests will run. If that all passes, I can accept it in to the main branch :)

**Pull request merged**